src/PortingAssistant.Client.Common/Utils/FileSystemAccess.cs

```diff
@@ -29,6 +29,27 @@ public static List<string> CheckWriteAccessForDirectory(string path)
             return result;
         }
 
+        /// <summary>
+        /// Checks csproj and at least one .cs file is writeable
+        /// </summary>
+        /// <param name="path">Project file path</param>
+        /// <returns>True if csproj and at least one .cs file is writeable</returns>
+        public static bool CheckWriteAccessForProject(string path)
+        {
+            return CanWriteFile(path) &&
+                 DirectoryHasWriteableCSharpFile(Path.GetDirectoryName(path));
+        }
+
+        private static bool DirectoryHasWriteableCSharpFile(string directoryPath)
+        {
+            bool fileFound = Directory.GetFiles(directoryPath)
+                .Where(file => Path.GetExtension(file) == ".cs")
+                .Any(file => !CanWriteFile(file));
+
+            return fileFound || 
+                Directory.GetDirectories(directoryPath).Any(subDirectory => DirectoryHasWriteableCSharpFile(subDirectory));
+        }
```

Comment:
Is the directorypath always the project path?

Reply:
No it is not always the project folder path. It should start as the project folder path, but then with the recursion become the subdirectories of the project folder and so on.

Is that ok?

Reply:
Wouldn't that be an issue for dirs without a cs file?

Reply:
Per our discussion, it will return false for dirs without a cs file, but should not be an issue as long as the project contains  a cs file somewhere within the project folder.